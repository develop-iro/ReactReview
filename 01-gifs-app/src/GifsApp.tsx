// Main app component for GIF search functionality
// Modules used:
// - useTranslation: React-i18next hook for internationalization
// - GifGrid: Displays a grid of GIFs
// - Header: Shows the app title and description
// - LanguageSwitcher: Allows user to change language
// - PreviousSearches: Shows and manages recent search terms
// - SearchInput: Input field for searching GIFs
// - useGifs: Custom hook to manage GIF search logic and state

import { useTranslation } from "react-i18next";
import { GifGrid } from "./components/GifGrid";
import { Header } from "./components/Header";
import { LanguageSwitcher } from "./components/LanguageSwitcher";
import { PreviousSearches } from "./components/PreviousSearches";
import { SearchInput } from "./components/SearchInput";
import { useGifs } from "./hooks/useGifs";

const GifsApp = () => {
  const { t } = useTranslation();
  const { gifResponse, previousSearches, handleSearch, handlePreviousSearch } =
    useGifs();

  return (
    <>
      <Header title={t("app.title")} description={t("app.description")} />
      <LanguageSwitcher />
      <SearchInput
        buttonText={t("app.search")}
        placeHolder={t("app.placeholder")}
        ariaLabel={t("app.search")}
        onSearch={handleSearch}
      />
      <PreviousSearches
        title={t("app.previousSearches")}
        searches={previousSearches}
        onSearchClick={handlePreviousSearch}
      />
      <GifGrid gifs={gifResponse} />
    </>
  );
};

export default GifsApp;
