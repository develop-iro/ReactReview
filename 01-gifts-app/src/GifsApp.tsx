import { useState } from "react";
import { useTranslation } from "react-i18next";
import { GifGrid } from "./components/GifGrid";
import { Header } from "./components/Header";
import { LanguageSwitcher } from "./components/LanguageSwitcher";
import { PreviousSearches } from "./components/PreviousSearches";
import { SearchInput } from "./components/SearchInput";
import { mockGifs } from "./data/mockData";

const searches = ["Superman", "Batman", "Spiderman", "Ironman"];

const GifsApp = () => {
  const { t } = useTranslation();
  const [searchTerm, setSearchTerm] = useState("");
  const [previousSearches] = useState([
    "Superman",
    "Batman",
    "Spiderman",
    "Ironman",
  ]);

  const handleSearch = (value: string) => {
    setSearchTerm(value);
  };

  const handlePreviousSearch = (search: string) => {
    setSearchTerm(search);
  };

  return (
    <>
      <Header title={t("app.title")} description={t("app.description")} />
      <LanguageSwitcher />
      <SearchInput
        placeHolder={t("app.placeholder")}
        ariaLabel={t("app.search")}
      />
      <PreviousSearches
        title={t("app.previousSearches")}
        searches={searches}
        onSearchClick={handlePreviousSearch}
      />
      <GifGrid gifs={mockGifs} />
    </>
  );
};

export default GifsApp;
