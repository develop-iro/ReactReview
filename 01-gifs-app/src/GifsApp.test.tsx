import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import GifsApp from "./GifsApp";

describe("GifsApp", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<GifsApp />);
    expect(asFragment()).toMatchSnapshot();
  });
});
