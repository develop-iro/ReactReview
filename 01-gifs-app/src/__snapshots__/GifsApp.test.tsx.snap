// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GifsApp > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="content-center"
  >
    <h1>
      app.title
    </h1>
    <p>
      app.description
    </p>
  </div>
  <div
    class="content-center"
    style="display: flex; flex-direction: row; justify-content: center; align-items: center;"
  >
    <button
      style="margin: 0px 8px; padding: 8px 16px; border-radius: 6px; background-color: rgb(25, 118, 210); color: white; font-size: 16px; cursor: pointer; transition: background 0.2s;"
    >
      English
    </button>
    <button
      style="margin: 0px 8px; padding: 8px 16px; border-radius: 6px; background-color: rgb(25, 118, 210); color: white; font-size: 16px; cursor: pointer; transition: background 0.2s;"
    >
      Espa√±ol
    </button>
  </div>
  <div
    class="search-container content-center"
  >
    <input
      aria-label="app.search"
      placeholder="app.placeholder"
      type="text"
      value=""
    />
    <button>
      app.search
    </button>
  </div>
  <div
    class="previous-searches"
  >
    <h2>
      app.previousSearches
    </h2>
    <ul
      class="previous-searches-list"
    />
  </div>
  <div
    class="gifs-container"
  />
</DocumentFragment>
`;
